### PR TITLE
Implement missed translations for anonymous `*`, `**` and `&` params

### PR DESCRIPTION
**Pull request by amomchilov**
### Motivation

Further additions to #51, #124, #161.

### CODE REVIEW

File: parser/prism/Translator.h [hunk 1]
```diff
@@ -19,6 +19,7 @@ class Translator final {
     // but don't have explicit ownership over it. We take a temporary reference to it, but we can't
     // escape that scope, which is why Translator objects can't be copied, or even moved.
     core::GlobalState &gs;
+    uint16_t uniqueCounter = 1;
```
Comment (amomchilov) [inline parser/prism/Translator.h:22]: See the `$` values in the test expectations, to see how this is used.

Comment (amomchilov) [reply parser/prism/Translator.h:22]: p.s. I noticed that parts (all of?) the rest of the repo uses a suffixed `_` for member variables. We should come back and change this, as well as any other such stylistic changes.

File: parser/prism/Translator.cc [hunk 2]
```diff
@@ -119,9 +119,17 @@ std::unique_ptr<parser::Node> Translator::translate(pm_node_t *node) {
...
-            std::string_view name = parser.resolveConstant(blockParamNode->name);
+            core::NameRef sorbetName;
+            if (auto prismName = blockParamNode->name; prismName != PM_CONSTANT_ID_UNSET) {
+                // A named block parameter, like `def foo(&block)`
+                auto name = parser.resolveConstant(prismName);
+                sorbetName = gs.enterNameUTF8(name);
+            } else { // An anonymous block parameter, like `def foo(&)`
+                sorbetName =
+                    gs.freshNameUnique(core::UniqueNameKind::Parser, core::Names::ampersand(), ++uniqueCounter);
+            }
```
Comment (egiurleo) [inline parser/prism/Translator.cc:130]: This all makes complete sense to me. Would it make sense to extract this into a `sorbetName` helper?

Comment (amomchilov) [reply parser/prism/Translator.cc:130]: There's a strange inconsistency with `parser::Restarg`, which needs the `nameLoc` to be modified conditionally :(

**Review by egiurleo — APPROVED** [review]
Sad times re: only one of the node types requiring a `nameLoc`. Go ahead and ship it and we'll figure it out!

**Pull request merged**